src/vs/workbench/api/common/extHostCommands.ts

```diff
@@ -219,6 +221,20 @@ export class ExtHostCommands implements ExtHostCommandsShape {
 		if (!command) {
 			throw new Error('Unknown command');
 		}
+		type ExtensionActionTelemetry = {
+			extensionId: string;
+			id: string;
+		};
+		type ExtensionActionTelemetryMeta = {
+			extensionId: { classification: 'SystemMetaData'; purpose: 'FeatureInsight'; owner: 'digitarald'; comment: 'The id of the extension handling the command, informing which extensions provide most-used functionality.' };
+			id: { classification: 'SystemMetaData'; purpose: 'FeatureInsight'; owner: 'digitarald'; comment: 'The id of the command, to understand which specific extension features are most popular.' };
+		};
+		if (command.extension && !command.extension.isBuiltin) {
+			this._telemetry.$publicLog2<ExtensionActionTelemetry, ExtensionActionTelemetryMeta>('Extension:ActionExecuted', {
+				extensionId: command.extension?.identifier.value,
+				id: id,
+			});
+		}
```

Comment:
I like the typings being defined locally, at the place of the usage but I would suggest to extract this into the some reportTelemetry-method which is called with the command in question